planner/core/plan_to_pb.go

```diff
@@ -159,11 +159,19 @@ func (p *PhysicalIndexScan) ToPB(ctx sessionctx.Context) (*tipb.Executor, error)
 			columns = append(columns, findColumnInfoByID(tableColumns, col.ID))
 		}
 	}
+	var pkColIds []int64
+	if p.Table.IsCommonHandle {
```

Comment:
Not all index scan need common handle, according to https://github.com/pingcap/tidb/blob/master/docs/design/2020-05-08-cluster-index.md#indexscan-schema

Better set a field in `PhyscialIndexScan` to indicate if common handle is needed.